code/game/machinery/machinery.dm

```diff
@@ -816,6 +816,33 @@ Class Procs:
 		return 1
 	return 0
 
+//exclusively for use with machines being made from the flatpacker
+//works like the parts exchange but because we only call this from a certain location
+//we can make some particular assumptions about what's going on
+/obj/machinery/proc/force_parts_transfer(var/datum/design/mechanic_design/O)
+	if(component_parts)
+		var/list/X = O.parts
+		var/M = X.Copy(1,0) //We paid for it when we made the flatpack, now we get new components
+		var/obj/item/weapon/circuitboard/CB = locate(/obj/item/weapon/circuitboard) in component_parts
+		var/P
+		for(var/obj/item/A in component_parts)
+			for(var/D in CB.req_components) //this logic is only necessary because some parts are unscannable - trade injector
+				if(ispath(A.type, D))
+					P = D
+					break
+			for(var/obj/item/B in M)
+				if(B.mech_flags != MECH_SCAN_FAIL) //prevent adding components that aren't scannable - they should already not be in this list, but I guess I'm paranoid?
```

Comment:
the same thing I said in the other file, compare with & and I suggest you avoid right creep